cmd/batch-handlers.go

```diff
@@ -70,10 +70,86 @@ type BatchJobRequest struct {
 	Started   time.Time            `yaml:"-" json:"started"`
 	Replicate *BatchJobReplicateV1 `yaml:"replicate" json:"replicate"`
 	KeyRotate *BatchJobKeyRotateV1 `yaml:"keyrotate" json:"keyrotate"`
-	Expire    *BatchJobExpire      `yaml:"expire" json:"expire"`
+	Expire    *BatchJobExpireV1    `yaml:"expire" json:"expire"`
 	ctx       context.Context      `msg:"-"`
 }
 
+// BatchJobRequestV2 this is an internal data structure not for external consumption.
+type BatchJobRequestV2 struct {
+	BatchJobRequest
+	Replicate *BatchJobReplicateV2 `yaml:"replicate" json:"replicate"`
+	Expire    *BatchJobExpireV2    `yaml:"expire" json:"expire"`
+}
+
+// BatchJobRequestVersion is indicating the version of the batch job request
+type BatchJobRequestVersion struct {
+	Replicate *struct {
+		APIVersion string `yaml:"apiVersion" json:"apiVersion"`
+	} `yaml:"replicate" json:"replicate"`
+	Expire *struct {
+		APIVersion string `yaml:"apiVersion" json:"apiVersion"`
+	} `yaml:"expire" json:"expire"`
+}
+
+// GetAPIVersion gets the api version
+func (b *BatchJobRequestVersion) GetAPIVersion() string {
+	if b.Replicate != nil {
+		return b.Replicate.APIVersion
+	}
+	if b.Expire != nil {
+		return b.Expire.APIVersion
+	}
+	return "v1"
+}
+
+func unmarshalBatchJobRequest(buf []byte) (job *BatchJobRequestV2, err error) {
+	job = &BatchJobRequestV2{}
+	version := &BatchJobRequestVersion{}
+	if err = yaml.Unmarshal(buf, version); err != nil {
+		return nil, err
+	}
+	switch version.GetAPIVersion() {
+	case "v1":
+		buf, err = batchJobRequestMigrateV1ToV2(buf)
+		if err != nil {
+			return nil, err
+		}
+		fallthrough
+	case "v2":
+		if err = yaml.Unmarshal(buf, job); err != nil {
+			return nil, err
+		}
+	}
+	return
+}
+
+func batchJobRequestMigrateV1ToV2(buf []byte) ([]byte, error) {
```

Comment:
You can directly return *BatchJobRequestV2 value after migrating from a v1 value. You don't have to marshal it to only unmarshal it as v2 again.
```suggestion
func batchJobRequestMigrateV1ToV2(buf []byte)(*BatchJobRequestV2, error) {
```